### ISSUE TITLE
Nested v-for breaks variable usage detection

### ISSUE THREAD

**Issue by HerringtonDarkholme** [vuejs/eslint-plugin-vue#164]
* **ESLint Version:** 4.5.0
* **eslint-plugin-vue Version:** 3.12.0
* **Node Version:** 8.0.0
 
```js
module.exports = {
  extends: [
    'eslint:recommended',
    'plugin:vue/recommended' // or 'plugin:vue/base'
  ],
}
```

Code:
```vue
<template>
  <div>
  <template v-for="item in dealItemList">
    <li v-for="snapItem in item.dealItemSnapList" :key="snapItem.id">
      123
    </li>
  </template>
  </div>
</template>
```

**What did you expect to happen?**

No error, since inner iterator `snapItem` depends on `item`, thus `snapItem.id` uses both inner and outer variables.

**What actually happened? Please include the actual, raw output from ESLint.**

Error.

**Comment by armano2** [comment]
confirmed:
```
severity: 'Error'
message: 'Expected 'v-bind:key' directive to use the variables which are defined by the 'v-for' directive. (vue/valid-v-for)'
at: '4,51'
source: 'eslint'
```
